bundles/http/Citrix/0b4d1a6b-30f3-4134-a354-42fa008b757e/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,1029 @@
+        let supervisorPersonNumber = data.SupervisorData ?.[0].Supervisor[0].$.PersonNumber ?? null;
+        let supervisorEmail = '';
+        if(supervisorPersonNumber != null)
+        {
+            let dt = personDetails.get(supervisorPersonNumber)
+            supervisorEmail = dt != undefined ? dt.EMailAddresses?.[0].EMailAddress[0].$.Address ?? '' : null
+        }
+        dataStore.save("personalDetails", {
+            "fullName": data.PersonData[0].Person?.[0].$.FullName ?? null,
+            "firstName": data.PersonData[0].Person?.[0].$.FirstName ?? null,
+            "lastName": data.PersonData[0].Person?.[0].$.LastName ?? null,
+            "email": data.EMailAddresses?.[0].EMailAddress[0].$.Address ?? null,
+            "personNumber": data.PersonData[0].Person?.[0].$.PersonNumber ?? null,
+            "supervisorName": data.SupervisorData?.[0].Supervisor[0].$.FullName ?? null,
+            "supervisorPersonNumber": supervisorPersonNumber,
+            "supervisorEmail": supervisorEmail
+        });
+    })
+    
+}
+
+async function syncAccrualBalance(dataStore, client, session, people,integrationParameters) {
+    const date = moment().format(integrationParameters.dateFormat);
+    const request = createRequest(client,session);
+    for (const pno of people) {
+        const body =  `<?xml version='1.0' encoding='UTF-8' ?>
+        <Kronos_WFC version="1.0">
+            <Request Action="Load">
+                <AccrualData>
+                    <BalanceDate>${date}</BalanceDate>
+                    <Employee>
+                        <PersonIdentity personNumber="${pno.personNumber}"/>
+                    </Employee>
+                </AccrualData>
+            </Request>
+        </Kronos_WFC>`
+        let {response,result} = await request(body)
+        let path = result.Kronos_WFC.Response[0].AccrualData[0].AccrualBalances[0].AccrualBalanceSummary ?? [];
+        let accrualData = path.map(value => {
+            return {
+                "accrualCodeId": value.$.AccrualCodeId,
+                "accrualCodeName": value.$.AccrualCodeName,
+                "accrualType": value.$.AccrualType,
+                "encumberedBalanceInTime": Number(value.$.EncumberedBalanceInTime),
+                "hoursPerDay": value.$.HoursPerDay,
+                "projectedVestedBalanceInTime": value.$.ProjectedVestedBalanceInTime,
+                "projectedDate": moment(value.$.ProjectedDate,`${integrationParameters.dateFormat}`).toDate(),
+                "projectedGrantAmountInTime": value.$.ProjectedGrantAmountInTime,
+                "projectedTakingAmountInTime": value.$.ProjectedTakingAmountInTime,
+                "vestedBalanceInTime": value.$.VestedBalanceInTime,
+                "personalNumber": pno.personNumber
+            }
+        });
+        dataStore.save("accrualBalances", accrualData);
+    }
+}
+
+async function syncRequestTimeOff(dataStore, client, session, people,integrationParameters) {
```

Comment:
We have merged both the functions into one, now on the basis of latestSynchronizationTime value it will determine the functionality. If the latestSynchronizationTime is null it behaves as full sync and if latestSynchronizationTime is not null it behaves as incremental sync